todo/forms.py

```diff
@@ -29,8 +33,17 @@ class AddEditTaskForm(ModelForm):
 
     def __init__(self, user, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        
+        # Django's ManyToManyRel is a little odd in that its directional sense 
+        # is not guranteeed and must be tested for. One of these models is User, 
+        # the other is the Groups model (the one group_field points to). We seek 
+        # the attribute with which we can access the set of users in a group.  
+        group_field = getattr(user._meta.model, setting("TODO_USER_GROUP_ATTRIBUTE"), "groups")
+        candidates = (group_field.rel.related_name, group_field.rel.field.attname)
+        user_attr = candidates[1] if group_field.rel.model == user._meta.model else candidates[0]
```

Comment:
I can almost but not quite track what this is doing. Can you think of a clearer way to write the python, or add another comment? 

It seems like it is having to recompute for every accessing user every time, whereas it's going to be a project-wide constant for everyone. That seems wasteful. 

Wait - Could this become simpler this way? I think we can assume that any alternate Groups system subclasses django Group. Therefore can't we do something like 

```
if not hasattr(Group, 'parent'):
    setting("TODO_USER_GROUP_ATTRIBUTE")
```

That's not complete but I think you see where I'm going with it - just detect whether the current Groups model has been subclassed.

Man you'd think Django would just have a setting for establishing an alternate Group model. Then it would be easy.

I've thrown out several ideas here, sorry it's not complete, just spitballing clearer and less expensive ways to do this.

Reply:
Wait! We don't need to detect `parent` attribute, and we don't need to inspect every users.  All you need to do is to have a shared/reusable function that resolves `TODO_USER_GROUP_ATTRIBUTE` into a real model. That's it - done! Should be just a line or two.

Reply:
All good. Indeed, this could be a generic Django utility function (should really be in Django models already. It's asking "What is the related model for this field?" and because the field is a Django relation it turns out that I can't find that answer in one place. There are two models stored in oddly inconsistent places in the relation object and which one is this model and which one is the related model seems to probably depend on which model was declared first at a guess! Not 100% sure, but it's not reliable)

Reply:
Would you mind if I gave it a try?

Reply:
Not sure what you mean, but no I don't mind ...